src/com/ensa/gi4/service/impl/GestionMaterielFacadeImpl.java

```diff
@@ -142,27 +82,22 @@ public void rendreMateriel(TypeMateriel type, String nomMateriel, int userId) {
 				break;
 			}
 		}
-		switch (type) {
-		case LIVRE:
-			serviceActuel = gestionLivreService;
-			break;
-		case CHAISE:
-			serviceActuel = gestionChaiseService;
-			break;
-		default:
-			System.out.println("Le type " + type + " n'est pas reconnu");
-			return;
-		}
+		assignTheAppropriateService(serviceActuel, type);
 		if (hasMateriel) {
 			serviceActuel.rendreMateriel(nomMateriel, userId);
-		}else {
+		} else {
 			System.out.println("vous pouvez pas rendre ce materiel");
 		}
 	}
 
 	@Override
 	public void markerNonDisponible(TypeMateriel type, String nomMateriel) {
-		final GestionMaterielService serviceActuel;
+		GestionMaterielService serviceActuel = null;
+		assignTheAppropriateService(serviceActuel, type);
+		serviceActuel.markerNonDisponible(nomMateriel);
+	}
+
+	private void assignTheAppropriateService(GestionMaterielService serviceActuel, TypeMateriel type) {
```

Comment:
re-assigned parameter is not used.

Reply:
```java
 public void f( A a ) {
a = something;
}
/// .... 
A aa = aThing;
f(aa);
assert aa == something ; // this will fail 
// in  other words, the f method has no effect on passed in references
```

Reply:
probleme fixed sir i did not pay attention to that